docs: Added release notes and associated media for 0.38

garden-of-delete opened the pull request:
### SUMMARY
We would like to publish release notes with each major release. This is a first crack at sorting out the format for the release notes, starting with v0.38. These are the decision points that would benefit from feedback or objections, and will likely see iteration in future release notes:
- Multimedia embedded in the release notes vs no multimedia
- A single .md file with each new version's notes appended vs a separate .md file and directory for each release
- Directory structure for release notes
- How the release notes are linked to the top level README.md

The changes in this PR make a suggestion for each of those points.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
N/A

### TEST PLAN
No testing required, no changes to codebase. 

### ADDITIONAL INFORMATION
<!--- Check any relevant boxes with "x" -->
<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->
- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

garden-of-delete committed:
Added release notes and associated media for 0.38

garden-of-delete committed:
Added ASF license header to release notes

benceorlai commented:
Thank Robert!

mistercrunch commented:
See it rendered here:
https://github.com/garden-of-delete/incubator-superset/blob/master/RELEASING/release-notes-0-38/release-notes-0-38.md

Comments:
- file could be `/release-notes-0-38/README.md` so that github renders it
- expecting to see images embedded in the release notes but I don't see any (?)

garden-of-delete committed:
refactored the release notes directory structure. Re-added media to the 0.38 release notes.

garden-of-delete committed:
fixed broken image and hyperlink

garden-of-delete committed:
fixed broken image and hyperlink pt 2

garden-of-delete commented:
> See it rendered here:
> https://github.com/garden-of-delete/incubator-superset/blob/master/RELEASING/release-notes-0-38/release-notes-0-38.md
> 
> Comments:
> 
> * file could be `/release-notes-0-38/README.md` so that github renders it when clicking on the folder
> * expecting to see images embedded in the release notes but I don't see any (?)

Addressed in most recent commits.

benceorlai commented:
@garden-of-delete Github's MD supports <img> html tags, which gives you better control over the layout of images. Could you try something like:

<img src="/media/10800.gif" style="max-width:50%; border:1px solid grey;">

garden-of-delete committed:
Swapped out media embed tags

Swapped out github style markdown embed tags with html style ones for more control.

garden-of-delete committed:
Fixing small inconsistencies

Fixed a couple small inconsistencies in the feature tags

garden-of-delete committed:
Resize media, update embed style

Seems like GitHub does not any but the most basic css in the embed tag.

etr2460 commented on RELEASING/release-notes-0-38/README.md:
```diff
@@ -0,0 +1,319 @@
+[List of backwards incompatible changes](https://github.com/apache/incubator-superset/blob/master/UPDATING.md#0380)
+
+## Complete Changelog
+For the complete chnagelog please see [apache/incubator-superset/CHANGELOG.md](https://github.com/apache/incubator-superset/blob/master/CHANGELOG.md)
```
spelling nit: changelog

garden-of-delete committed:
Spelling nit

mistercrunch commented:
Super nit-picky, but can we add borders to images? https://stackoverflow.com/questions/37349314/is-it-possible-to-add-border-to-image-in-github-markdown

Looks a bit rough here:
https://github.com/garden-of-delete/incubator-superset/tree/master/RELEASING/release-notes-0-38/

ktmud commented on RELEASING/release-notes-0-38/README.md:
```diff
@@ -0,0 +1,319 @@
+
+Database list view refactor/redesign ([Roadmap](https://github.com/apache-superset/superset-roadmap/issues/55))
+- feat: database delete warning  (#[10800](https://github.com/apache/incubator-superset/pull/10800))
+<img src="media/10800.gif" width="50%"/>
```
Could you indent the image three spaces so it's wrapped under the list item?

garden-of-delete replied:
images are now wrapped under the bullet points.

garden-of-delete committed:
Fixed some formatting nits

Added a box around the images to clearly distinguish them from the whitespace. Images are now embedded on the same indentation tier as their associated bullet point.

mistercrunch commented:
Trying the old close/reopen trick since Cypress is not complying

garden-of-delete committed:
Rebase to hopefully fix cypres problems

garden-of-delete committed:
May the cypress gods be pleased with this commit.

robdiciuccio commented:
Nit: would love to see `alt` tags on the images for improved accessibility.
